include/swift/AST/ASTBridging.h

```diff
@@ -2653,58 +2656,105 @@ SWIFT_NAME("getter:BridgedPattern.boundName(self:)")
+  BridgedLayoutConstraintKindUnknownLayout,
+  BridgedLayoutConstraintKindTrivialOfExactSize,
+  BridgedLayoutConstraintKindTrivialOfAtMostSize,
+  BridgedLayoutConstraintKindTrivial,
+  BridgedLayoutConstraintKindClass,
+  BridgedLayoutConstraintKindNativeClass,
+  BridgedLayoutConstraintKindRefCountedObject,
+  BridgedLayoutConstraintKindNativeRefCountedObject,
+  BridgedLayoutConstraintKindBridgeObject,
+  BridgedLayoutConstraintKindTrivialStride,
 };
 
+class BridgedLayoutConstraint {
+  swift::LayoutConstraintInfo *_Nullable raw;
+
+public:
+  SWIFT_UNAVAILABLE("Use the factory methods")
+  BRIDGED_INLINE BridgedLayoutConstraint();
+
+  SWIFT_UNAVAILABLE("Use the factory methods")
+  BRIDGED_INLINE BridgedLayoutConstraint(swift::LayoutConstraint constraint);
+
+  BRIDGED_INLINE swift::LayoutConstraint unbridged() const;
+};
+
+SWIFT_NAME("BridgedLayoutConstraint.getLayoutConstraint(_:id:)")
+BridgedLayoutConstraint
+BridgedLayoutConstraint_getLayoutConstraint(BridgedASTContext cContext,
+                                            BridgedIdentifier cID);
+
+SWIFT_NAME("BridgedLayoutConstraint.getLayoutConstraint(_:kind:)")
+BridgedLayoutConstraint
+BridgedLayoutConstraint_getLayoutConstraint(BridgedASTContext cContext,
+                                            BridgedLayoutConstraintKind cKind);
+
+SWIFT_NAME(
+    "BridgedLayoutConstraint.getLayoutConstraint(_:kind:size:alignment:)")
+BridgedLayoutConstraint
+BridgedLayoutConstraint_getLayoutConstraint(BridgedASTContext cContext,
+                                            BridgedLayoutConstraintKind cKind,
+                                            size_t size, size_t alignment);
+
+SWIFT_NAME("getter:BridgedLayoutConstraint.isNull(self:)")
+BRIDGED_INLINE bool
+BridgedLayoutConstraint_isNull(BridgedLayoutConstraint cConstraint);
+
+SWIFT_NAME("getter:BridgedLayoutConstraint.kind(self:)")
+BridgedLayoutConstraintKind
+BridgedLayoutConstraint_getKind(BridgedLayoutConstraint cConstraint);
+
+SWIFT_NAME("getter:BridgedLayoutConstraint.isKnownLayout(self:)")
+BRIDGED_INLINE bool
+BridgedLayoutConstraint_isKnownLayout(BridgedLayoutConstraint cConstraint);
+
+SWIFT_NAME("getter:BridgedLayoutConstraint.isTrivial(self:)")
+BRIDGED_INLINE bool
+BridgedLayoutConstraint_isTrivial(BridgedLayoutConstraint cConstraint);
+
 enum ENUM_EXTENSIBILITY_ATTR(open) BridgedRequirementReprKind : size_t {
-  /// A type bound T : P, where T is a type that depends on a generic
-  /// parameter and P is some type that should bound T, either as a concrete
-  /// supertype or a protocol to which T must conform.
   BridgedRequirementReprKindTypeConstraint,
-
-  /// A same-type requirement T == U, where T and U are types that shall be
-  /// equivalent.
   BridgedRequirementReprKindSameType,
-
-  /// A layout bound T : L, where T is a type that depends on a generic
-  /// parameter and L is some layout specification that should bound T.
   BridgedRequirementReprKindLayoutConstraint,
```

Comment:
Is removing the comments here intentional? If so, should they also be removed from `MacroDefinitionKind`? I don't really feel too strongly one way or the other, but I think we ought to be consistent

Reply:
Will do. I don't think we need duplicated comments for each bridging decls.

Reply:
Oh wait. I see there's no corresponding `swift::` namespace C++ definition of `MacroDefinitionKind`. I'm leaving comments on `MacroDefinitionKind`.